server/portal/libs/agave/operations.py

```diff
@@ -333,6 +333,19 @@ def copy(client, src_system, src_path, dest_system, dest_path, file_name=None,
     return copy_result
 
 
+def makepublic(client, src_system, src_path, dest_path='/', *args, **kwargs):
+    dest_system = next((sys['system']
+                        for sys in settings.PORTAL_DATAFILES_STORAGE_SYSTEMS
+                        if sys['scheme'] == 'public'))
+
+    return copy(client,
```

Comment:
This is more of a comment about the general deficiency we have in system to system file operations, currently. We can spin this off into another ticket rather than addressing here if you like.

By returning copy (which [calls client.files.importData](https://github.com/TACC/Frontera-Portal/blob/c05fed0da01a1c2203220b51b4d83735e5153319/server/portal/libs/agave/operations.py#L317)), this appears to be a non-blocking call. The reference code for agavepy shows them [wrapping the response in an AgaveAsyncResponse object](https://github.com/TACC/agavepy/blob/02a3874c2ddea3a87022bb0c429587973ec56dd8/tests/test_003_integration.py#L359), which blocks the call until the copy operation is complete.

In general, I think this is a deficiency with system to system copy operations that can be solved by:
- Registering a webhook callback ([example](https://github.com/TACC/Frontera-Portal/blob/300421e05585889259b7dcad9715e82c3e97b1a6/server/portal/apps/system_creation/utils.py#L42))
- [providing a callbackURL argument for importData](https://agavepy.readthedocs.io/en/master/docsite/files/files.html#importdata-import-a-file-via-direct-upload-or-importing-from-a-url-to-the-user-s-default-storage-location)

Reply:
Made a ticket for this: https://jira.tacc.utexas.edu/browse/FP-863